src/arkode/arkode_sprkstep_io.c

```diff
@@ -0,0 +1,463 @@
+                                  &ark_mem, &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  if (onoff)
+  {
+    arkSetUseCompensatedSums(arkode_mem, SUNTRUE);
+    ark_mem->step = sprkStep_TakeStep_Compensated;
+    if (!step_mem->yerr) 
+    {
+      if (!arkAllocVec(ark_mem, ark_mem->yn, &(step_mem->yerr)))
+      {
+        return ARK_MEM_FAIL;
+      }
+    }
+  }
+  else
+  {
+    arkSetUseCompensatedSums(arkode_mem, SUNFALSE);
+    ark_mem->step = sprkStep_TakeStep;
+  }
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  SPRKStepSetMethod:
+
+  Specifies the SPRK method
+
+  ** Note in documentation that this should not be called along
+  with SPRKStepSetOrder. **
+  ---------------------------------------------------------------*/
+int SPRKStepSetMethod(void* arkode_mem, ARKodeSPRKStorage sprk_storage)
+{
+  ARKodeMem ark_mem          = NULL;
+  ARKodeSPRKStepMem step_mem = NULL;
+  int retval                 = 0;
+
+  /* access ARKodeSPRKStepMem structure */
+  retval = sprkStep_AccessStepMem(arkode_mem, "SPRKStepSetMethod", &ark_mem,
+                                  &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  if (step_mem->method)
+  {
+    ARKodeSPRKStorage_Free(step_mem->method);
+    step_mem->method = NULL;
+  }
+
+  step_mem->method = sprk_storage;
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  SPRKStepSetMethodName:
+
+  Specifies the SPRK method.
+  ---------------------------------------------------------------*/
+int SPRKStepSetMethodName(void* arkode_mem, const char* method)
+{
+  ARKodeMem ark_mem          = NULL;
+  ARKodeSPRKStepMem step_mem = NULL;
+  int retval                 = 0;
+
+  /* access ARKodeSPRKStepMem structure */
+  retval = sprkStep_AccessStepMem(arkode_mem, "SPRKStepSetMethodName", &ark_mem,
+                                  &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  if (step_mem->method)
+  {
+    ARKodeSPRKStorage_Free(step_mem->method);
+    step_mem->method = NULL;
+  }
+
+  step_mem->method = ARKodeSPRKStorage_LoadByName(method);
+
+  return step_mem->method ? ARK_SUCCESS : ARK_ILL_INPUT;
+}
+
+/*---------------------------------------------------------------
+  SPRKStepSetOrder:
+
+  Specifies the method order
+
+  ** Note in documentation that this should not be called along
+  with SPRKStepSetMethod. **
+  ---------------------------------------------------------------*/
+int SPRKStepSetOrder(void* arkode_mem, int ord)
+{
+  ARKodeMem ark_mem          = NULL;
+  ARKodeSPRKStepMem step_mem = NULL;
+  int retval                 = 0;
+
+  /* access ARKodeSPRKStepMem structure */
+  retval = sprkStep_AccessStepMem(arkode_mem, "SPRKStepSetOrder", &ark_mem,
+                                  &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  if (ord == 7 || ord == 9 || ord > 10) { return ARK_ILL_INPUT; }
```

Comment:
This is slightly inconsistent with the documentation. If the user previously changed the order to a valid option then later tries to change the order to an illegal the previously set value will be used instead of the default.

Reply:
So should it behave like the documentation says or should it stay as-is and the documentation updated?